src/decider/gatewaydecider/utils.rs

```diff
@@ -2259,21 +2283,27 @@ pub async fn get_unified_key(
             let key_prefix = C::GLOBAL_LEVEL_OUTAGE_KEY_PREFIX;
             let base_key = if payment_method_type == CARD {
                 vec![
-                    key_prefix,
-                    &payment_method_type,
-                    &payment_method,
-                    gateway_scoring_data.bankCode.as_deref().unwrap_or(""),
-                    gateway_scoring_data.cardType.as_deref().unwrap_or(""),
+                    key_prefix.to_string(),
+                    payment_method_type.to_string(),
+                    payment_method.to_string(),
+                    gateway_scoring_data.bankCode.as_deref().unwrap_or("").to_string(),
+                    gateway_scoring_data.cardType.as_deref().unwrap_or("").to_string(),
                 ]
             } else if payment_method_type == UPI {
+                let peek = gateway_scoring_data
+                    .paymentSource
+                    .as_ref()
+                    .map(|s| s.peek().to_string())
+                    .unwrap_or_default();
```

Comment:
```suggestion
                let peek = gateway_scoring_data.get_payment_source();
```
add a impl for `get_payment_source` in gateway_scoring_data and avoid duplication in other places